### ISSUE TITLE
inconsistent results when wrapping with inject_hyperparams 

### ISSUE THREAD

**Issue by aorvieto** [google-deepmind/optax#583]
I noticed in quite a few projects of mine that my results change when initializing the optimizer with

`tx =  optax.inject_hyperparams(optax.adamw)(learning_rate=opt.lr, weight_decay=opt.wd)`

as opposed to

`tx =  optax.adamw(learning_rate=opt.lr, weight_decay=opt.wd)`

The learning rate here is constant, and what follows in the code is the usual initialization of the TrainState, followed by the training loop.

I did not find online an explanation for this phenomenon – and is definitely a source for errors.

Thank you
Antonio

**Comment by mtthss** [comment]
Can you share a small repro? E.g. in a colab?

**Comment by mtthss** [comment]
Any update on this?

**Comment by fabianp** [comment]
gentle ping @aorvieto . If the problem does not happen anymore please close the issue :-)

**Comment by aorvieto** [comment]
Hi all! I struggled to find the time to provide a small working example - I
saw this happening only on large models. I can follow up on this in a few
weeks

Il giorno dom 3 dic 2023 alle ore 16:28 Fabian Pedregosa <
***@***.***> ha scritto:

**Comment by fabianp** [comment]
Closing as need more info. Please do reopen when we have a reproducible test :-)
